### PR TITLE
[Unity][FX] Add support for PT2.0 scaled_dot_product_attention

### PR DESCRIPTION
**Pull request by masahi**
diffusers started to use `scaled_dot_product_attention` as of v0.16 in SD VAE. See the doc https://pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html. We cannot support attention mask, dropout, and causal mask optimization for now. 

The PT attention op requires a different input format than our attention op, so we need to transpose inputs. Luckily those transpose can be canceled in practice, since diffusers does transpose on q/k/v before calling `scaled_dot_product_attention` anyway (a design mistake?): https://github.com/huggingface/diffusers/blob/909742dbd6873052995dc6cd5f4150ff238015d2/src/diffusers/models/attention_processor.py#L906-L908

I'm also doing clean up on FX test cases. We shouldn't need `@tvm.testing.requires_gpu` since the tests don't execute anything on GPU. I'll also try removing local `import torch` if possible (CI should have PT installed, right?)

@MasterJH5574 @jinhongyii  @cyx-6

### CODE REVIEW

**Comment by yongwww** [comment]
@masahi torch is installed in CI GPU container, but not in CI CPU container.

**Comment by masahi** [comment]
> @masahi torch is installed in CI GPU container, but not in CI CPU container.

Oh I thought the cpu image has PT as well. It is a bit ironic since even for GPU image we install CPU-build of PT (since we only use PT for reference). 

Is there a reason the CPU image cannot have PT?

**Comment by yongwww** [comment]
> Oh I thought the cpu image has PT as well. It is a bit ironic since even for GPU image we install CPU-build of PT (since we only use PT for reference).
> 
> Is there a reason the CPU image cannot have PT?

I also noticed this before, don't know the reason why pt cpu version was specified explicitly.

**Comment by masahi** [comment]
> I also noticed this before, don't know the reason why pt cpu version was used in gpu container.

Because their gpu build often causes a trouble when we upgrade the PT version in our CI. 

It seems our cpu image has tflite, mxnet, and caffe installed. I see no reason PT cannot be installed. I'll work on that next week.

**Comment by yongwww** [comment]
> > I also noticed this before, don't know the reason why pt cpu version was used in gpu container.
> 
> Because their gpu build often causes a trouble when we upgrade the PT version in our CI.
> 
> It seems our cpu image has tflite, mxnet, and caffe installed. I see no reason PT cannot be installed. I'll work on that next week.

Just tried to install torch 2.0 in my local cpu container (run from the same image as CI), it works well

**Comment by masahi** [comment]
Okay after https://github.com/apache/tvm/pull/14842 lands and the new image is published, I'll update our cpu image.

**Comment by vinx13** [comment]
note that float mask that's added to score can be supported via `bias` input of relax attention

**Review by yongwww — APPROVED** [review]
LGTM

**Comment by masahi** [comment]
Cleaned up test cases quite a bit.

**Review by vinx13 — APPROVED** [review]
